src/Features/Core/Portable/SpellCheck/AbstractSpellCheckSpanService.cs

```diff
@@ -165,20 +164,48 @@ private void AddStringSubSpans(SyntaxToken token)
                     var spanEnd = currentChar.Span.End;
 
                     var seenEscape = false;
-                    while (
-                        currentCharIndex < virtualChars.Length &&
-                        virtualChars[currentCharIndex] is { IsLetter: true } endChar)
+                    while (currentCharIndex < virtualChars.Length)
                     {
-                        // we know if we've seen a letter that is an escape character if it takes more than two actual
-                        // characters in the source.
-                        seenEscape = seenEscape || endChar.Span.Length > 1;
-                        spanEnd = endChar.Span.End;
-                        currentCharIndex++;
+                        var endChar = virtualChars[currentCharIndex];
+                        if (IsWordPart(endChar))
+                        {
+                            // we know if we've seen a letter that is an escape character if it takes more than two actual
+                            // characters in the source.
+                            seenEscape = seenEscape || endChar.Span.Length > 1;
+                            spanEnd = endChar.Span.End;
+                            currentCharIndex++;
+                        }
+                        else if (endChar == ' ' && endChar.Span.Length == 1)
+                        {
+                            // Consume a regular space (common between words) to keep the number of spans we report low.
+                            spanEnd = endChar.Span.End;
+                            currentCharIndex++;
```

Comment:
added this optimization as well.  so we don't generate a high amount of word spans when there's just a single escape in the string.